pages/OTPScreen.js

```diff
@@ -31,7 +31,14 @@ const OTPScreen = ({
             setError(true);
             return;
         }
-        verifyOtp(otp);
+        setLoading(true)
+        verifyOtp(otp)
+            .then(() => {
+                setLoading(false);
+            })
+            .catch ((error) => {
+                setLoading(false);
+            });
```

Comment:
Use `.finally()`.
Both happy and error paths do the same thing. In such cases, use `finally`